src/metadata.rs

```diff
@@ -158,17 +167,45 @@ pub struct Group {
     pub members: Vec<GroupMember>
 }
 
+fn parse_member_assignment(payload_rdr: &mut Cursor<&[u8]>) -> Result<Vec<MemberAssignment>> {
+    let _version = payload_rdr.read_i16::<BigEndian>().chain_err(|| "Failed to parse value version")?;
+    let assign_len = payload_rdr.read_i32::<BigEndian>().chain_err(|| "Failed to parse assigment length")?;
+    let mut assigns = Vec::with_capacity(assign_len as usize);
+    for _ in 0..assign_len {
+        let topic = read_str(payload_rdr).chain_err(|| "Failed to parse assignment topic name")?.to_owned();
+        let partition_len = payload_rdr.read_i32::<BigEndian>().chain_err(|| "Failed to parse assignment partition length")?;
+        let mut partitions = Vec::with_capacity(partition_len as usize);
+        for _ in 0..partition_len {
+            let partition = payload_rdr.read_i32::<BigEndian>().chain_err(|| "Failed to parse assignment partition")?;
+            partitions.push(partition);
+        }
+        assigns.push(MemberAssignment {
+            topic: topic,
+            partitions: partitions
+        })
+    }
+    Ok(assigns)
+}
+
 fn fetch_groups(consumer: &MetadataConsumer, timeout_ms: i32) -> Result<Vec<Group>> {
     let group_list = consumer.fetch_group_list(None, timeout_ms)
         .chain_err(|| "Failed to fetch consumer group list")?;
 
     let mut groups = Vec::new();
     for rd_group in group_list.groups() {
         let members = rd_group.members().iter()
-            .map(|m| GroupMember {
-                id: m.id().to_owned(),
-                client_id: m.client_id().to_owned(),
-                client_host: m.client_host().to_owned()
+            .map(|m| {
+                let mut assigns = vec![];
+                if let Some(assignment) = m.assignment() {
```

Comment:
We should have an additional check here to only parse the field if the `protocol_type` is the expected one. Some users might use Kafka's group membership for other kind of groups, and cause the parser to fail or to return unexpected results.